pyigm/surveys/igmsurvey.py

```diff
@@ -231,6 +243,103 @@ def add_abs_sys(self, abs_sys):
         # Append
         self._abs_sys.append(abs_sys)
 
+    def build_all_abs_sys(self, linelist=None, **kwargs):
+        """
+        Build all of the AbsSystem objects from the _dict
+        or _data if the _dict does not exist!
+        In that order
+        """
+        # This speeds things up a bunch
+        if linelist is None:
+            linelist = default_linelist(self.verbose)
+        # Loop me
+        if len(self._dict) > 0:
+            print("Starting the AbsSystem build for the _dict.  Be patient..")
+            for key in self._dict.keys():
+                _ = self.build_abs_sys_from_dict(key, linelist=linelist, **kwargs)
+        elif len(self._data) > 0:
+            for qq in range(self.nsys):
+                _ = self.build_abs_sys_from_data(qq)
+        else:
+            raise IOError("Nothing to build the systems with!")
+        # Return
+        print("Done!")
+        return
+
+    def build_abs_sys_from_data(self, row):
+        """ Build an AbsSystem from the _data
+        The item in self._abs_sys is filled and
+        the system is also returned
+
+        Parameters
+        ----------
+        row : int
+          Row of the _data table
+          Ignores any masking -- this may change
+
+        Returns
+        -------
+        abs_sys : AbsSystem
+
+        """
+        # vlim -- may make optional
+        vlim=self._data['vlim'][row]
+        if self._data['vlim'].unit is not None:
+            vlim *= self._data['vlim'].unit
+        else:
+            vlim = vlim * u.km/u.s
+        # skwargs
+        skwargs = {}
+        for key in ['NHI', 'sig_NHI', 'name', 'zem']:
+            if key in self._data.keys():
+                skwargs[key] = self._data[key][row]
+        # Instantiate
+        abssys = class_by_type(self.abs_type)(self.coords[row], self._data['zabs'][row], vlim, **skwargs)
+        # Fill
+        if len(self._abs_sys) == 0:
+            self.init_abs_sys()
+        self._abs_sys[row] = abssys
+        # Return too
+        return abssys
+
+    def build_abs_sys_from_dict(self, abssys_name, **kwargs):
```

Comment:
Just curious, why are you not using the linetools.isgm.AbsSystem.from_dict() class method?  Since IGMSystem is just subclassed from AbsSystem, I think at least some rationale is warranted in the docstring.

Reply:
Great catch!  I have to add an input change to 
AbsSystem.from_dict() but that is a good idea.